### PR TITLE
#182300803 Authentication check middlewares-

### PR DESCRIPTION
**Pull request by Himbaza123h**
## What does this PR do?
1.  It is a middleware that check if the user is authenticated. and assign user with response if Authorized elsewhere display an error
 ## More descriptions about this task.
1. Middleware Authentication to verify token
2. Helper to generate token
3. Routes to work within your controllers
## How should it be manually tested?
1.  Clone this repo into your local computer
2. Create a `.env` file referring to the `.env.example` file in the root folder and provide values for those keys.
3. Run `npm` install for installing all dependencies.
4. Run `npm run dev` to start server
5.  Check by any routes  with token and without token to see if there is difference by providing error  
## Screenshoot if appropriate?
1.  N/A
## What are the relevant pivotal tracker stories?
Story type: `feature(feat)`
StoryID:  `182300803`

### CODE REVIEW

**Comment by habaumugisha1** [comment]
can you flow the git naming convention for PR and commit message should be formatted well as manager shared in yesterday's stundup?

**Comment by habaumugisha1** [comment]
Honore, why are you pushing commented codes? like in controller you comment everything

File: src/utils/helpers/AuthToken.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { status } from '../express/lib/response';
+import { sign, verify } from 'jsonwebtoken';
+import { config } from 'dotenv';
+config();
+
+class AuthToken{
+    static TokenGenerator(data){
+        const token=sign({
+            data
```
Comment (habaumugisha1) [inline src/utils/helpers/AuthToken.js]: can you write horizontally?

**Review by nyakamwe — APPROVED** [review]

**Comment by Himbaza123h** [comment]
> can you flow the git naming convention for PR and commit message should be formatted well as manager shared in yesterday's stundup?

Sure sir! I'm working based on each feedback

**Comment by Himbaza123h** [comment]
> Honore, why are you pushing commented codes? like in controller you comment everything

> I worked on it, later I discovered that it was not supposed to be a part of my task, that why I commented it.

File: src/utils/helpers/AuthToken.js (hunk 1 above)
Comment (Himbaza123h) [reply src/utils/helpers/AuthToken.js]: > can you write horizontally?

Sure sir! noted.

**Review by pacomesimon — COMMENTED** [review]
Regarding the lines you commented because they didn't belong to your task, I think that it may affect code clarity when this branch will be merged with develop.
So, I suggest that, if it is possible, you would delete the commits exclusively for those commented files (in your branch), so that they won't affect the code clarity. (I think I have to research if such commits deletions can be done in git)

**Comment by Himbaza123h** [comment]
> Regarding the lines you commented because they didn't belong to your task, I think that it may affect code clarity when this branch will be merged with develop. So, I suggest that, if it is possible, you would delete the commits exclusively for those commented files (in your branch), so that they won't affect the code clarity. (I think I have to research if such commits deletions can be done in git)

Okay I get your point! I'm on it

**Pull request merged**